frontend/src/pages/UserDetails.tsx

```diff
@@ -59,10 +84,15 @@ const UserDetailsPage: React.FC = () => {
 
   return (
     <div className="mt-24 min-h-screen w-full p-4">
+      <canvas ref={canvasRef} style={{ display: 'none' }}></canvas>
       <div className="mx-auto md:max-w-3xl">
         <div className="overflow-hidden rounded-3xl bg-white shadow-xl dark:bg-gray-800">
           <div className="relative">
-            <div className="h-32 bg-owasp-blue"></div>
+            <div className="h-32 bg-gray-800 dark:bg-black">
+              {imageLink && (
+                <img src={imageLink} className="h-full w-full object-cover object-[54%_60%]" />
```

Comment:
Showing a loading indicator while the image is being generated would be very user friendly.